### PR TITLE
.NET Interactive: Add_try_dotnet_batch_29

### PR DESCRIPTION
**Pull request by v-mepa**
fyi @mairaw 

**Add_try_dotnet_batch_29**

URL | PageViews | BounceRate | Try .NET Status | Comments
-- | -- | -- | -- | --
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.distinct | 11335 | 11 | Batch 29 | First example only. Others   depend on each other
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.any | 9971 | 11 | Batch 29 | First example only. Others are   not runnable.
https://docs.microsoft.com/en-us/dotnet/api/system.decimal.round | 6583 | 11 | Batch 29 | Fix output from sample 3
https://docs.microsoft.com/en-us/dotnet/api/system.globalization.datetimeformatinfo | 4999 | 8 | Batch 29 | except sample 7, 8, 9, 16, 17,   18 / fix output for sample 11, 12
https://docs.microsoft.com/en-us/dotnet/api/system.exception.innerexception | 4803 | 55 | Batch 29 | Fix sample output
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.count | 3931 | 11 | Batch 29 | First example only; remaining sample is not   runnable as-is
https://docs.microsoft.com/en-us/dotnet/api/system.string.toupper | 3784 | 12 | Batch 29 | First example only; remaining sample is culture   dependent
https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.match | 3724 | 11 | Batch 29 | Fix sample output in the first two examples
https://docs.microsoft.com/en-us/dotnet/api/system.outofmemoryexception | 3360 | 13 | Batch 29 | Second sample only. Remaining samples don't work   on try .net
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.sum | 2760 | 13 | Batch 29 | Only until you get to the overloads containing   Func. Also remove the Examples section entirely for content not containing   Func if they have a warning that the code example uses an overload that is   different from the specific overload that this topic describes.
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.max | 2377 | 12 | Batch 29 | Only until you get to the overloads containing   Func. Also remove the Examples section entirely for content not containing   Func if they have a warning that the code example uses an overload that is   different from the specific overload that this topic describes.
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.defaultifempty | 2183 | 9 | Batch 29 | Only last example; others are not runnable as-is
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.average | 1915 | 15 | Batch 29 | Remove the Examples section entirely for content   not containing Func if they have a warning that the code example uses an   overload that is different from the specific overload that this topic   describes.
https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.matchcollection | 1832 | 9 | Batch 29 | Fix sample output
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.elementat | 1742 | 10 | Batch 29 | Improve sample output message
https://docs.microsoft.com/en-us/dotnet/api/system.char.toupper | 1504 | 11 | Batch 29 | Second sample output needs to be fixed
https://docs.microsoft.com/en-us/dotnet/api/system.uint16.maxvalue | 817 | 22 | Batch 29 |  
https://docs.microsoft.com/en-us/dotnet/api/system.timespan.zero | 801 | 7 | Batch 29 |  
https://docs.microsoft.com/en-us/dotnet/api/system.decimal.truncate | 794 | 10 | Batch 29 |  
https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.groupcollection | 629 | 10 | Batch 29 | Fix   sample output (and perhaps review brands used)

### CODE REVIEW

File: xml/System.Linq/Enumerable.xml [hunk 1]
```diff
@@ -2405,9 +2335,9 @@ Dim doubles = From item As Integer In sequence
 ## Examples  
  The following code example demonstrates how to use <xref:System.Linq.Enumerable.Count%60%601%28System.Collections.Generic.IEnumerable%7B%60%600%7D%2CSystem.Func%7B%60%600%2CSystem.Boolean%7D%29> to count the elements in an array that satisfy a condition.  
   
- [!code-csharp[System.Linq.Enumerable#23](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Linq.Enumerable/CS/enumerable.cs#23)]
- [!code-vb[System.Linq.Enumerable#23](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Linq.Enumerable/VB/Enumerable.vb#23)]  
-  
+ :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Linq.Enumerable/CS/enumerable.cs" interactive="try-dotnet-method" id="Snippet23":::
```
Comment (mairaw) [inline xml/System.Linq/Enumerable.xml:2338]:
```suggestion
 :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Linq.Enumerable/CS/enumerable.cs" id="Snippet23":::
```

File: xml/System.Linq/Enumerable.xml [hunk 2]
```diff
@@ -2487,14 +2417,14 @@ Dim doubles = From item As Integer In sequence
   
  This example uses a non-empty sequence.  
   
- [!code-csharp[System.Linq.Enumerable#24](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Linq.Enumerable/CS/enumerable.cs#24)]
- [!code-vb[System.Linq.Enumerable#24](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Linq.Enumerable/VB/Enumerable.vb#24)]  
-  
+ :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Linq.Enumerable/CS/enumerable.cs" interactive="try-dotnet-method" id="Snippet24":::
```
Comment (mairaw) [inline xml/System.Linq/Enumerable.xml:2420]:
```suggestion
 :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Linq.Enumerable/CS/enumerable.cs" id="Snippet24":::
```

File: xml/System.Linq/Enumerable.xml [hunk 3]
```diff
@@ -820,9 +810,9 @@
 ## Examples
```
Comment (mairaw) [inline xml/System.Linq/Enumerable.xml:810]: For the Enumerable.Average, the Func examples can also be interactive.

Also, lines 745-753 should be removed. This is still showing up:
image

**Review by mairaw — CHANGES_REQUESTED** [review]
Thanks @v-mepa. Changes look good but we need a couple of fixes in before merging.

**Comment by mairaw** [comment]
I've applied most of my suggestions already except for one that needs to be done manually. Make sure to do a git pull before making the manual changes.

File: xml/System.Linq/Enumerable.xml (hunk 3 above)
Comment (v-mepa) [reply xml/System.Linq/Enumerable.xml:810]: For applying interactive in Func examples the following comment will apply?
_"Remove the Examples section entirely for content not containing Func if they have a warning that the code example uses an overload that is different from the specific overload that this topic describes."_

File: xml/System.Linq/Enumerable.xml [hunk 4]
```diff
@@ -1933,9 +1854,9 @@
   
  [!INCLUDE[sqo_diff_overload_example_func](~/includes/sqo-diff-overload-example-func-md.md)]  
   
- [!code-csharp[System.Linq.Enumerable#18](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Linq.Enumerable/CS/enumerable.cs#18)]
- [!code-vb[System.Linq.Enumerable#18](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Linq.Enumerable/VB/Enumerable.vb#18)]  
-  
+ :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Linq.Enumerable/CS/enumerable.cs" interactive="try-dotnet-method" id="Snippets18":::
+ :::code language="vb" source="/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Linq.Enumerable/VB/Enumerable.vb" id="Snippet18":::
```
Comment (mairaw) [inline xml/System.Linq/Enumerable.xml:1858]:
```suggestion
 :::code language="vb" source="~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Linq.Enumerable/VB/Enumerable.vb" id="Snippet18":::
```

File: xml/System.Linq/Enumerable.xml [hunk 5]
```diff
@@ -1933,9 +1854,9 @@
   
  [!INCLUDE[sqo_diff_overload_example_func](~/includes/sqo-diff-overload-example-func-md.md)]  
   
- [!code-csharp[System.Linq.Enumerable#18](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Linq.Enumerable/CS/enumerable.cs#18)]
- [!code-vb[System.Linq.Enumerable#18](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Linq.Enumerable/VB/Enumerable.vb#18)]  
-  
+ :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Linq.Enumerable/CS/enumerable.cs" interactive="try-dotnet-method" id="Snippets18":::
```
Comment (mairaw) [inline xml/System.Linq/Enumerable.xml:1857]:
```suggestion
 :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Linq.Enumerable/CS/enumerable.cs" interactive="try-dotnet-method" id="Snippet18":::
```

**Review by mairaw — APPROVED** [review]
LGTM. My latest change should fix the warning.

We just need to verify the C# example is being displayed for https://review.docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.average?branch=pr-en-us-5265&view=net-5.0#System_Linq_Enumerable_Average__1_System_Collections_Generic_IEnumerable___0__System_Func___0_System_Single__ after the build finishes.

**Pull request merged**